pandas/core/indexes/datetimes.py

```diff
@@ -589,8 +589,8 @@ def get_loc(self, key, method=None, tolerance=None):
         if is_valid_nat_for_dtype(key, self.dtype):
             key = NaT
 
-        if isinstance(key, self._data._recognized_scalars):
-            # needed to localize naive datetimes
+        if isinstance(key, (self._data._recognized_scalars, date)):
```

Comment:
pls don't do this - you can maybe modify the accepted scalars instead

Reply:
Totally makes sense, since the entire refactoring was about cleaning up these situations. Moved to the _recognized_scalars.